test/built-ins/Set/prototype/union/allows-set-like-class.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2023 Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set.prototype.union
+description: GetSetRecord allows Set-like objects
+info: |
+    1. If obj is not an Object, throw a TypeError exception.
+    2. Let rawSize be ? Get(obj, "size").
+    ...
+    7. Let has be ? Get(obj, "has").
+    ...
+    9. Let keys be ? Get(obj, "keys").
+features: [Set-methods]
+includes: [compareArray.js]
+---*/
+
+const s1 = new Set([1, 2]);
+const s2 = {
+  size: 2,
+  has: function () {
+    throw new Test262Error("Set.prototype.union should not invoke .has on its argument");
+  },
+  keys: function* keys() {
+    yield 2;
+    yield 3;
+  },
+};
```

Comment:
I'm afraid you went a bit too far here - this is now basically a duplicate of `test/built-ins/Set/prototype/union/allows-set-like-object.js`, while the file name suggests that it should cover a class